tests/featured-speaker.spec.ts

```diff
@@ -1,21 +1,35 @@
+        tagLine: 'Dynamic PR Specialist and Communications Expert',
     },
-].forEach(({ name, bio }) => {
-    test(`Should provide link to page of featured speaker ${name} on home page`, async ({ page }) => {
+    {
+        fullName: 'Top Speaker 3',
+        bio: 'Top Speaker 3 is a seasoned software engineer with a deep expertise in full-stack development. With over 15 years in the tech industry, they have contributed to numerous high-profile projects, specializing in creating robust and scalable web applications. Known for their innovative problem-solving skills and dedication to continuous learning, Top Speaker 3 is a frequent speaker at international tech conferences. Outside of work, they are passionate about coffee, gourmet cooking, and exploring the latest trends in technology.',
+        tagLine: 'Innovative Tech Visionary and Mentor',
+    },
+    {
+        fullName: 'Top Speaker 4',
+        bio: 'Top Speaker 4 is a highly regarded web development expert with a passion for pioneering new technologies. With over a decade of experience in the tech industry, they have been at the forefront of numerous innovative projects, pushing the boundaries of what is possible on the web. Known for their dynamic approach and ability to solve complex problems, Top Speaker 4 is a sought-after speaker at tech conferences worldwide. When not coding, they enjoy exploring new tech trends, mentoring aspiring developers, and indulging in their love for gourmet cuisine.',
+        tagLine: 'Pioneering Innovator in Tech Solutions',
+    },
+].forEach(({ fullName, tagLine, bio }) => {
+    test(`Should provide link to page of featured speaker ${fullName} on home page`, async ({ page }) => {
         await page.goto('/');
-        await page.getByRole('menu', { name: 'Featured Speakers' }).getByRole('link', { name: name }).click();
-        await expect(page.getByRole('heading', { name: name })).toBeVisible();
+        const featuredSpeakersSection = page.getByRole('region', { name: 'Featured Speakers' });
+        const featuredSpeakersMenu = featuredSpeakersSection.getByRole('menu', { name: 'Featured Speakers' });
+        const featuredSpeakersMenuItem = featuredSpeakersMenu.getByRole('menuitem', { name: fullName });
+        await expect(featuredSpeakersMenuItem.getByText(fullName)).toBeVisible();
+        await expect(featuredSpeakersMenuItem.getByText(tagLine)).toBeVisible();
+        await featuredSpeakersMenuItem.getByRole('link', { name: fullName }).click();
```

Comment:
I would say this could already be a separate test. We don't test here whether the link was provided; we check if the link is working, no?

Reply:
I think it depends. Other than with unit or component tests, my understanding of the best practices of e2e tests is to test whole features of the website. More precise, multiple assertions are totally fine, because tools like Playwright or Cypress can give very precise information about failed assertions. And more tests slow down the tests, whereas multiple assertions in on test are fast.  In this case, the feature is to provide a link to every featured speaker with some information. And, of course, the link should also work. 

TLDR: From my understanding, the check if the link is working should not be separated. 😅

---

However, I check here if the full bio is available on the speaker's page. That's not really part of this feature. In other tests, I check just the URL.